core/src/main/java/org/bptlab/cepta/operators/SumOfDelayAtStationFunction.java

```diff
@@ -18,42 +19,49 @@
     It will return a Stream of Tuple2 with the location Id and the sum of delay. 
     */
-    public DataStream<Tuple2<Long, Double>> SumOfDelayAtStation(DataStream<TrainDelayNotification> inputStream, int windowSize) {
+    public DataStream<Tuple2<Long, Double>> SumOfDelayAtStation(DataStream<T> inputStream, int windowSize, String locationAttributName) {
         DataStream<Tuple2<Long, Double>> resultStream = inputStream
         .keyBy(
-            new KeySelector<TrainDelayNotification, Integer>(){
+            new KeySelector<T, Integer>(){
                 Integer key = Integer.valueOf(0);
-                public Integer getKey(TrainDelayNotification event){
+                public Integer getKey(T event){
                     Integer returnKey = key/windowSize;
                     key++;
                     return returnKey;
-                } 
+                }
             }
         )
         .window(GlobalWindows.create()).trigger(
             CustomCountTrigger.of(windowSize)
         ).process(
-            sumOfDelayAtStationWindowProcessFunction()
+            sumOfDelayAtStationWindowProcessFunction(locationAttributName)
         );
         return resultStream;
-    }; 
+    };
 
-    public static ProcessWindowFunction<TrainDelayNotification, Tuple2<Long, Double>, Integer, GlobalWindow> sumOfDelayAtStationWindowProcessFunction() {
-        return new ProcessWindowFunction<TrainDelayNotification, Tuple2<Long, Double>, Integer, GlobalWindow>() {
+    public ProcessWindowFunction<T, Tuple2<Long, Double>, Integer, GlobalWindow> sumOfDelayAtStationWindowProcessFunction(String locationAttributName) {
+        return new ProcessWindowFunction<T, Tuple2<Long, Double>, Integer, GlobalWindow>() {
             @Override
-            public void process(Integer key, Context context, Iterable<TrainDelayNotification> input, Collector<Tuple2<Long, Double>> out) throws Exception {
+            public void process(Integer key, Context context, Iterable<T> input, Collector<Tuple2<Long, Double>> out) throws Exception {
                 HashMap<Long, Double> sums = new HashMap<Long, Double>();
-                for (TrainDelayNotification in: input) {
-                    Long trainId = in.getTrainId();
-                    Long locationId = in.getLocationId();
-                    Double delay = in.getDelay();
+                for (T in: input) {
+                    Class c = in.getClass();
+                    String methodName = "get" + locationAttributName;
+                    Method method = c.getDeclaredMethod(methodName);
+                    Long locationId = Long.valueOf(method.invoke(in).toString());
+                    method = c.getDeclaredMethod("getDelay");
+                    Double delay = Double.valueOf(method.invoke(in).toString());
+
+                   // Long trainId = in.getTrainId();
+                   // Long locationId = in.getLocationId();
+                   // Double delay = in.getDelay();
```

Comment:
Is this comment needed? Looks like its from some previous iteration

Reply:
no